pentaho/impl/src/main/java/pt/webdetails/cdf/dd/CdeSettings.java

```diff
@@ -151,57 +140,54 @@ public static CdfDDSettings getSettings(){
 
     return files.toArray( new IBasicFile[ files.size() ] );
   }
-  
-  public static String getEncoding(){
+
+  public static String getEncoding() {
     return CharsetHelper.getEncoding();
   }
-  
+
   public static class CdfDDSettings extends PluginSettings {
 
-    public CdfDDSettings( IRWAccess writeAccess ){
+    public CdfDDSettings( IRWAccess writeAccess ) {
       super( writeAccess ); // useful when unit testing / mocking
     }
 
-    public CdfDDSettings(){
-      super(CdeEnvironment.getPluginSystemWriter());
+    public CdfDDSettings() {
+      super( CdeEnvironment.getPluginSystemWriter() );
     }
-    
-    List<Element> getComponentLocationElements(){
-      return getSettingsXmlSection("custom-components/path");
+
+    List<Node> getComponentLocationElements() {
+      return getSettingsXmlSection( "custom-components/path" );
```

Comment:
Just downcast?

Reply:
In dom4j 2.x, the class inheritance changed.

Reply:
What do you mean? `Element` is still a subtype of `Node` (http://javadoc.io/doc/org.dom4j/dom4j/2.1.1)? It's always been like that I suppose...

Again, you should just downcast.

Also, it does not make sense to return a list of nodes in a method that says it returns a list of elements in its name.